### ISSUE TITLE
Google not loaded

### ISSUE THREAD

**Issue by faichelbaum** [aptly-io/grav-plugin-googlemaps#4]
Hi

I followed your howto but, the maps does not render.
I have the div created with no issue but in the Dev Console, I see the following message :
Uncaught ReferenceError: google is not defined

As I check the files loaded by the page, I can confirm that no JS from Google about the maps is loaded.
Any idea ?

**Comment by aptly-io** [comment]
Hi,

It's difficult to guess what might go on/wrong with the given details.

Is it possible send me the html page source as received by your browser 
as email attachment?

 From firefox or Chrome, that is control-U (on Linux):
- firefox' menu: Tools -> Web Developer -> Page Source
- chrome's menu: Tools -> View Source
  Likely similar for Safari and IE (the control key might be different on 
  other OS as well).

Hope this info can help me further.

Thanks

On 2016-04-02 12:07 PM, Francois Aichelbaum wrote:

> Hi
> 
> I followed your howto but, the maps does not render.
> I have the div created with no issue but in the Dev Console, I see the
> following message :
> Uncaught ReferenceError: google is not defined
> 
> As I check the files loaded by the page, I can confirm that no JS from
> Google about the maps is loaded.
> Any idea ?
> 
> ---
> 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/aptly-io/grav-plugin-googlemaps/issues/4

## 

Kind regards,
francis                   [home page](https://aptly.io)

"Empty cans make the most noise" - Farah Ann Abdul Hadi

**Comment by faichelbaum** [comment]
No worries. You can even test it on the dev site : 
https://dev.aichelbaum.com/my-trips

**Comment by aptly-io** [comment]
Hi,

I had a look at the generated HTML.
It misses some script includes that the googlemaps plug-in should have 
generated.
(<script src="https://maps.googleapis.com/maps/api/js?v=3&language=ja" 
type="text/javascript" ></script>
<script src="/user/plugins/googlemaps/assets/js/googlemaps.min.js" 
type="text/javascript" ></script>)

Could you clear the cache and then reload the page in the browser? From 
the command line it's this command:
bin/grav clear

If that did not help, my next question is if you did do something 
special with grav's caching?
Like setting specific options?
And you're using a recent version of Grav (>1.0.0)?

Is it a modular page? Can you show me (all) the header options for that 
mytrips page?

One last question what happens when you add the following header to that 
page
(not as a sub header of googlemaps but at the top level)?
cache_enable: false

Thanks
francis

On 2016-04-02 01:38 PM, Francois Aichelbaum wrote:

> No worries. You can even test it on the dev site :
> 
> ---
> 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/aptly-io/grav-plugin-googlemaps/issues/4#issuecomment-204699127

## 

Kind regards,
francis                   [home page](https://aptly.io)

"Empty cans make the most noise" - Farah Ann Abdul Hadi

**Comment by faichelbaum** [comment]
I did try with the cache being cleared earlier with no luck . Just retried : no luck yet
I disabled the cache : still the same result
As PHP is setup with APCu, I restarted it but no luck

So far, I have no idea. I can add the missing calls in the template but it's not the best solution.

**Comment by aptly-io** [comment]
Hi,

Regarding my other questions in my reply (see next), can you reply and 
try (do a bin/grav clear)?

(Is it a modular page? Can you show me (all) the header options for that 
mytrips page?

One last question what happens when you add the following header to that 
page
(not as a sub header of googlemaps but at the top level in mytrips 
page)?
cache_enable: false
)

Thanks

francis

On 2016-04-02 10:55 PM, Francois Aichelbaum wrote:

> I did try with the cache being cleared earlier with no luck . Just
> retried : no luck yet
> I disabled the cache : still the same result
> As PHP is setup with APCu, I restarted it but no luck
> 
> So far, I have no idea. I can add the missing calls in the template
> but it's not the best solution.
> 
> ---
> 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/aptly-io/grav-plugin-googlemaps/issues/4#issuecomment-204803592

## 

Kind regards,
francis                   [home page](https://aptly.io)

"Empty cans make the most noise" - Farah Ann Abdul Hadi

**Comment by faichelbaum** [comment]
So :

bin/grav clear => no change (done multiple times)

is the page a modular one => yes
header options (for the modular) :
title: 'My trips'
published: true
external_links:
    process: true
    title: false
    no_follow: true
    mode: active
    target: _blank
body_classes: modular
visible: true
content:
    order:
        dir: desc
        by: default
    items: '@self.modular'

headers for the text page:
title: 'Traveling so far ....'
external_links:
    process: true
    title: false
    no_follow: true
    mode: active
    target: _blank
googlemaps:
    enabled: true
    built_in_css: false
    trips:
        center: '51.009314, 4.061254'
        zoom: 12
        type: TERRAIN
        kmlStatus: false
        markers:
            -
                location: '51.009358, 4.061578'
                title: 'The local church'
                zIndex: 1
                timeout: 1000
                info: '<strong>The local church</strong>.<br/>Cleaned up in recent years!'
            -
                location: '51.017227, 4.073198'
                title: 'A secret passage'
                zIndex: 2
                timeout: 2000
                icon: 'https://aptly.io/user/pages/02.about/02.hiking/trail_nextto_railway_thumbnail.png'
                link: '#treasures'

adding the cache_enable: false to mytrips (btw, already globally disabled) : no change

**Comment by aptly-io** [comment]
Hi,

I don't see any issue with these header settings.
Unfortunately I've currently no idea how to provide you with a solution 
:-(
I look into this with a PHP debugger when I have time.

Thanks
franics

On 2016-04-03 10:09 AM, Francois Aichelbaum wrote:

> So :
> 
> bin/grav clear => no change (done multiple times)
> 
> is the page a modular one => yes
> header options (for the modular) :
> title: 'My trips'
> published: true
> external_links:
>     process: true
>     title: false
>     no_follow: true
>     mode: active
>     target: _blank
> body_classes: modular
> visible: true
> content:
>     order:
>         dir: desc
>         by: default
>     items: '@self.modular'
> 
> headers for the text page:
> title: 'Traveling so far ....'
> external_links:
>     process: true
>     title: false
>     no_follow: true
>     mode: active
>     target: _blank
> googlemaps:
>     enabled: true
>     built_in_css: false
>     trips:
>         center: '51.009314, 4.061254'
>         zoom: 12
>         type: TERRAIN
>         kmlStatus: false
>         markers:
>             -
>                 location: '51.009358, 4.061578'
>                 title: 'The local church'
>                 zIndex: 1
>                 timeout: 1000
>                 info: '<strong>The local church</strong>.<br/>Cleaned
> up in recent years!'
>             -
>                 location: '51.017227, 4.073198'
>                 title: 'A secret passage'
>                 zIndex: 2
>                 timeout: 2000
>                 icon:
> 'https://aptly.io/user/pages/02.about/02.hiking/trail_nextto_railway_thumbnail.png'
>                 link: '#treasures'
> 
> adding the cache_enable: false to mytrips (btw, already globally
> disabled) : no change
> 
> ---
> 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/aptly-io/grav-plugin-googlemaps/issues/4#issuecomment-204911198

## 

Kind regards,
francis                   [home page](https://aptly.io)

"Empty cans make the most noise" - Farah Ann Abdul Hadi

**Comment by aptly-io** [comment]
Hi,

TL;DR
Below I describe 2 minimal ways in which I test the googlemaps plugin 
(still a lot to write down and read for you) that work for me. Can you 
replicate one (or both) of these?

I'm afraid the only way to debug you case is to zip a 'minial set-up' 
that does not work for you. Send that to me. This will allow me to 
investigate/debug locally to see what goes on.
In case such minimal set-up would work for me, I'm afraid you'll have to 
consult the people on the grav mailing list for support on what might be 
wrong/different in your environment. Note both cases are on Ubuntu 
Linux.

(Note that another user narrowed down an issue between the assets and 
googlemaps plug-in. Are you using the assets plug-in? I'll provide a fix 
for that soonish).
1. Case 1: On virtual box running with the apache web server (a LAMP 
   stack).

It would get too length to explain virtual box installation (since I did 
this long time ago, I forgot T_T).

I have a $HOME directory on that virtual OS.
I have a local, normal user account (with admin sudo rights).
That account has a subdirectory $HOME/public_html in its home directory.

cd $HOME/public_html                          # go into that 
subdirectory
git clone https://github.com/getgrav/grav.git # get the pure grav source 
tree
cd grav                                       # move into the checked 
out grav original source tree
git checkout tags/1.0.10   # change to the last official release source 
code version: 1.0.10
bin/grav install                              # download all requires 
external sources
bin/gpm install admin                         # install the admin 
plug-in (add its dependencies)

cp htaccess.txt .htaccess                     # create a .htaccess

# With an editor Comment out the existing RewriteBase rule

# And add a new line in this .htaccess. The new line should contain

this:
RewriteBase /~<your-username-here>/grav/
               ^^^^^^^^^^^^^^^^^^^^

# Replace <your-username-here> with the actual username.

touch user/config/security.yaml               # create an empty 
security.yaml file
sudo chgrp -R www-data .                      # change group ownership 
to www-data

# The change to group owner www-data is to make sure the apache server

has the right

# to read the files in the home directory's subdirectory.

# Note once running and serving out files, apache will create it's own

files (like cache etc.)

# These new files will have his ownership, not your user account's.

# These files you can only remove if you have sudo rights.

Now that the sources are all installed, open a browser and open the web 
page. For me the virtual box guest runs on this IP address: 
http://192.168.56.110 (not accessible from internet). If all goes right, 
a generic apache page should appear for this URL.

Now go to           http://192.168.56.110/~<your-username-here>/grav
You should see grav's welcome page

Go to the admin page, you need to create a manager account first

Install the googlemaps plug-in from within admin. Even though there's a 
validation failure, the plug-in gets installed.

Change the default grav home page by going into expert mode.
(http://192.168.56.110/~<your-username-here>/grav/admin/pages/home/mode:expert)

In the headers add the following yaml config for the googlemaps plug-in:
googlemaps:
     tagid:
         center: '51.009314, 4.061254'
         zoom: 12
         type: TERRAIN

In the content pane, for example on the first line add this:
[GOOGLEMAPS:tagid]

Save and go to the home page again; I can see the map being drawn:
http://192.168.56.110/~<your-username-here>/grav

# In case apache does not show anything, or returns some error, look

into its log files.

# While running the apache and going to the troubling web page run this

cd /var/log/apache2                    # where apache keeps its log 
files
tail -f error.log                      # dump lines to the screen 
continuously

# In another console the same now for access.log

cd /var/log/apache2
tail -f access.log

These should give a hint what might be wrong serving grav ...
1. Case 2 Run locally directly under PHP build-in server

Even though much simpler, this is not adviced by grav's author and he 
even build-in nagging code that you have to comment out (see below). My 
local php has not issues with this mode of operation however. !!! Never 
use this to serve your home page on the Internet (for security reasons) 
!!! It's perfect for debugging PHP however.

php --version                            # my php version on Ubuntu 
15.10
PHP 5.6.11-1ubuntu3.1 (cli)

cd /tmp                                  # install all sources in /tmp
                                          # get official grav release 
package 1.0.10
wget 
https://github.com/getgrav/grav/releases/download/1.0.10/grav-admin-v1.0.10.zip
unzip grav-admin-v1.0.10.zip             # unpack archive
cd grav-admin                            # go inside root of unpacked 
grav source code

# edit this file by commenting out a nagging check on line 247, 248 and

249
vi /tmp/grav-admin/user/plugins/admin/admin.php

# Original lines:

```
         if (php_sapi_name() == 'cli-server') {
             throw new \RuntimeException('The Admin Plugin cannot run 
```

on the PHP built-in webserver. It needs Apache, Nginx or another 
full-featured web server.', 500);
             }

# Comment these out between /\*  ...  */

```
         /* if (php_sapi_name() == 'cli-server') {
             throw new \RuntimeException('The Admin Plugin cannot run 
```

on the PHP built-in webserver. It needs Apache, Nginx or another 
full-featured web server.', 500);
             }*/

php -S localhost:8000                    # run php's build-in server 
from the root

# browse to http://localhost:8000

You're presented with the admin-create-user-page. Add a user, email and 
its password. Then you're inside the dashboard. Add the googlemaps 
plugin and edit the home page as described in Case 1. (Even though 
there's a validation failure, the plug-in gets installed).

Go to the home page editor expert mode 
(http://localhost:8000/admin/pages/home/mode:expert). And insert again 
the marker and headers as described in Case 1

In the headers add the following YAML config lines (e.g. below the 
existing `title`):
googlemaps:
     tagid:
         center: '51.009314, 4.061254'
         zoom: 12
         type: TERRAIN

In the content pane, on the first line add this:
[GOOGLEMAPS:tagid]

Click save and go to the home page (http://localhost:8000/). For me the 
googlemaps appears ...

thanks
francis

On 2016-04-03 10:09 AM, Francois Aichelbaum wrote:

> So :
> 
> bin/grav clear => no change (done multiple times)
> 
> is the page a modular one => yes
> header options (for the modular) :
> title: 'My trips'
> published: true
> external_links:
>     process: true
>     title: false
>     no_follow: true
>     mode: active
>     target: _blank
> body_classes: modular
> visible: true
> content:
>     order:
>         dir: desc
>         by: default
>     items: '@self.modular'
> 
> headers for the text page:
> title: 'Traveling so far ....'
> external_links:
>     process: true
>     title: false
>     no_follow: true
>     mode: active
>     target: _blank
> googlemaps:
>     enabled: true
>     built_in_css: false
>     trips:
>         center: '51.009314, 4.061254'
>         zoom: 12
>         type: TERRAIN
>         kmlStatus: false
>         markers:
>             -
>                 location: '51.009358, 4.061578'
>                 title: 'The local church'
>                 zIndex: 1
>                 timeout: 1000
>                 info: '<strong>The local church</strong>.<br/>Cleaned
> up in recent years!'
>             -
>                 location: '51.017227, 4.073198'
>                 title: 'A secret passage'
>                 zIndex: 2
>                 timeout: 2000
>                 icon:
> 'https://aptly.io/user/pages/02.about/02.hiking/trail_nextto_railway_thumbnail.png'
>                 link: '#treasures'
> 
> adding the cache_enable: false to mytrips (btw, already globally
> disabled) : no change
> 
> ---
> 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/aptly-io/grav-plugin-googlemaps/issues/4#issuecomment-204911198

## 

Kind regards,
francis                   [home page](https://aptly.io)

"Empty cans make the most noise" - Farah Ann Abdul Hadi

**Comment by faichelbaum** [comment]
Hi Francis

I'll give those a shot tomorrow and will give you the result.
Thanks again.

Cheers.

**Comment by aptly-io** [comment]
I've closed the issue for lack of feedback.
